### DISCUSSION TITLE
How to create a shared library with maven?

### DISCUSSION THREAD

**Discussion by azat-dev** [micronaut-projects/micronaut-core#11406] [Q&A]
Hi! I use micronaut in multiple microservices, I want to extract shared functionality (Factories, Singletones etc) in a separate library, which will be used across all microservices?

I've tried to make it as a vanila maven repository but it seems doesn't see factories and beans inside it.

Is there any good example?

Thank you :)

**Comment by graemerocher** [comment]
Seems like missing documentation in the Maven plugin //cc @alvarosanchez

**Reply by alvarosanchez** [reply]
In fact, what we really need is a Micronaut Library Parent POM
